internal/upgradecheck/http.go

```diff
@@ -163,7 +159,15 @@ func CheckForUpgradesScheduler(versionString string, channel chan bool) {
 		}
 	}()
 
-	info, err := checkForUpgrades(versionString, backoff)
+	// Make sure cache is synced before using the client
+	for {
```

Comment:
:question: Is the `for` loop needed here?  From what I can tell `WaitForCacheSync()` will simply block and not return until the cache finally syncs, but just let me know if there is anything I'm missing.

Reply:
As far as I can tell, yeah, `WaitForCacheSync` is well named -- I can remove the loop, it makes it cleaner without, I just put it in because it felt a little odd to run a function without checking the return.

Actually, the more I look at this without any for loop, I wonder about the off chance that syncing will encounter an error: if we're in a loop, then we just keep erroring until we don't. I almost feel like this would be another good place for a retry with an exponential backoff. (The other solution is to error and exit the process; an operator in that condition would never check for an upgrade.)

My feeling right now is... maybe remove the for-loop, and create a ticket for further enhancements?

Reply:
My thought is that if `WaitForCacheSync()` returns `false`, we would simply log an error and then _not_ proceed with running the "check for upgrade" logic.

My thinking here is that the `manager` will also fail to start in the event the caches don't sync successfully, so we're probably looking at a failure to start PGO as a whole anyway (and therefore likely a container restart).